sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -205,20 +206,19 @@ static void CleanupThread()
     }
 
     // loop the connection pool index - Note: lock is re-taken for the mutex
-    // Notes:
-    // - If all connections are removed from a host-index, the index is not removed from the pool to
-    // reduce the time of the entire routine.
-    // - A host-index with size 0 (no connections) is ignored.
+    // Notes: The size of each host-index is always expected to be greater than 0 because the
+    // host-index is removed anytime it becomes empty.
     for (auto index = CurlConnectionPool::g_curlConnectionPool.ConnectionPoolIndex.begin();
-         index != CurlConnectionPool::g_curlConnectionPool.ConnectionPoolIndex.end()
-         && index->second.size() > 0;
-         index++)
+         index != CurlConnectionPool::g_curlConnectionPool.ConnectionPoolIndex.end();
+         index = index->second.size() == 0
+             ? CurlConnectionPool::g_curlConnectionPool.ConnectionPoolIndex.erase(index)
+             : ++index)
     {
-      // Pool index with waiting connections. Each pool index behaves as a Last-in-First-out list,
-      // the last connection moved to the pool will be the first to be re-used. Because of this, the
-      // oldest connection in the pool can be found at the end of the list. Looping the connection
-      // pool backwards until a connection that is not expired is found or until all connections are
-      // removed.
+      // Each pool index behaves as a Last-in-First-out (connections are added to the pool with
+      // push_front). The last connection moved to the pool will be the first to be re-used. Because
```

Comment:
We plan to change this to not use `push_front` in a separate PR right? It's good to keep it as-is for now.

Reply:
naaa,, I think the implementation is fine. There's no real benefit updating this to work inverted, so, I would just keep it like this